src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1421,92 +1424,152 @@ bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,
                                                 DWORD& rgb) noexcept
 {
     bool foundRGB = false;
+    bool isSharpSignFormat = false;
+    size_t rgbHexDigitCount = 0;
     bool foundValidColorSpec = false;
-    std::array<unsigned int, 3> colorValues = { 0 };
+    std::array<float, 3> colorValues = { 0 };
+    std::array<unsigned int, 3> parameterValues = { 0 };
     bool success = false;
-    // We can have anywhere between [11,15] characters
-    // 9 "rgb:h/h/h"
-    // 12 "rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 9 || string.size() > 12)
-    {
-        return false;
-    }
+    const auto stringSize = string.size();
 
-    // Now we look for "rgb:"
+    // First we look for "rgb:"
     // Other colorspaces are theoretically possible, but we don't support them.
-
     auto curr = string.cbegin();
     if ((*curr++ == L'r') &&
         (*curr++ == L'g') &&
         (*curr++ == L'b') &&
         (*curr++ == L':'))
     {
+        // We can have one of the following formats:
+        // 9 "rgb:h/h/h"
+        // 12 "rgb:hh/hh/hh"
+        // 15 "rgb:hhh/hhh/hhh"
+        // 18 "rgb:hhhh/hhhh/hhhh"
+        // Any other cases will be invalid.
+        if (!(stringSize == 9 || stringSize == 12 || stringSize == 15 || stringSize == 18))
+        {
+            return false;
+        }
+
+        foundRGB = true;
+        rgbHexDigitCount = ((stringSize - 3) / 3) - 1;
+    }
+
+    // Try the sharp sign format.
+    curr--;
+    if (*curr++ == L'#')

```

Comment:
This also doesn't look right - I don't think this is the right place to add this check. You could be half way through the `rgb` prefix before encountering a `#`, and that would still be considered valid. So `rgb#fff` parses as red, and `rgb#fffffffff` parses as yellow (although this is actually overflowing the buffer, so I suspect it could sometimes crash).

Reply:
Yeah it is fragile. I’ll try to make it better.